bin/prettier.js

```diff
@@ -368,6 +377,24 @@ function writeOutput(result) {
 }
 
 function eachFilename(patterns, callback) {
+  // The ignorer will be used to filter file paths after the glob is checked,
+  // before any files are actually read
+  const ignoreFilePath = options.ignorePath || DEFAULT_IGNORE_FILENAME;
+  const absoluteIgnoreFilePath = path.isAbsolute(ignoreFilePath)
+    ? ignoreFilePath
+    : path.resolve(process.cwd(), ignoreFilePath);
+  let ignoreText = "";
+
+  try {
+    ignoreText = fs.readFileSync(absoluteIgnoreFilePath, "utf8");
+  } catch (readError) {
+    if (readError.code !== FILE_NOT_FOUND_ERROR_CODE) {
+      throw readError;
+    }
+  }
+
+  const ignorer = ignore().add(ignoreText.split(/\r?\n/));
```

Comment:
That `.split` will create an array `[ "" ]` if `ignoreText` is `""`.

Might need to do this:

```js
const ignorer = ignore();

if (ignoreText) {
  ignorer.add(ignoreText.split(/\r?\n/));
}
```